src/Shared/Resource.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+namespace Doofinder\Shared;
+
+use Doofinder\Configuration;
+use Doofinder\Shared\Exceptions\ApiException;
+use Doofinder\Shared\Interfaces\HttpClientInterface;
+use Doofinder\Shared\Interfaces\HttpResponseInterface;
+use Doofinder\Shared\Interfaces\ModelInterface;
+
+abstract class Resource
+{
+    protected $httpClient;
+    protected $config;
+    protected $baseUrl;
+
+    /**
+     * @param HttpClientInterface $httpClient
+     * @param Configuration $config
+     */
+    protected function __construct(HttpClientInterface $httpClient, Configuration $config)
+    {
+        $this->httpClient = $httpClient;
+        $this->config = $config;
+        $this->baseUrl = $config->getBaseUrl();
+    }
+
+    /**
+     * @return Configuration
+     */
+    public function getConfig()
+    {
+        return $this->config;
+    }
+
+    /**
+     * @param string $url
+     * @param string $method
+     * @param class-string<ModelInterface> $model
+     * @param array $params
+     * @param array $headers
+     * @return HttpResponseInterface
+     * @throws ApiException
+     */
+    protected function requestWithJwt($url, $method, $model = null, array $params = [], array $headers = [])
+    {
+        /** @var HttpResponseInterface $response */
+        $response = $this->httpClient->request(
+            $url,
+            $method,
+            $params,
+            array_merge($headers, ["Authorization: Token {$this->config->getToken()}"])
```

Comment:
this authentication method isn't JWT

Reply:
@sonic182 I think @agutierrezrodriguez say to use Jwt instead this, because is more secure authentication.

Reply:
We don't have jwt, he means the function name

Reply:
We don't have Jwt but we can generate it. https://docs.doofinder.com/api/management/v2/index.html#section/Basics/Authentication

Reply:
I was meant that the function name doesn't explain the code. The used authenticated method isn't jwt. We should implement jwt because is more secure, but at the moment, to not delay more the PR, continue with Token auth and change the function name.